packages/roosterjs-content-model/lib/modelApi/common/mergeModel.ts

```diff
@@ -289,23 +295,34 @@ function applyDefaultFormat(
                         applyDefaultFormat(segment, format, applyDefaultFormatOption);
                     }
 
-                    segment.format =
-                        applyDefaultFormatOption == 'mergeAll'
-                            ? { ...format, ...segment.format }
-                            : {
-                                  ...format,
-                                  ...getSemanticFormat(segment),
-                              };
+                    segment.format = mergeSegmentFormat(
+                        applyDefaultFormatOption,
+                        format,
+                        segment.format
+                    );
                 });
                 break;
         }
     });
 }
 
-function getSemanticFormat(segment: ContentModelSegment): ContentModelSegmentFormat {
+function mergeSegmentFormat(
+    applyDefaultFormatOption: 'mergeAll' | 'keepSourceEmphasisFormat',
```

Comment:
Hmm good point, I think for now we can keep it like this, just for consistency with other APIs options. For example, in [iterateSelections](https://github.com/microsoft/roosterjs/blob/2c44dc5406799bd6647ccd69a443bc614766d55e/packages/roosterjs-content-model/lib/modelApi/selection/iterateSelections.ts#LL20C1-L20C91)

Reply:
Normally, if a type is used in multiple places, we should add a type. If it is only used by one public function, it is ok not to add a type.

Adding a string literal type is ok, but do not add enum. I'm trying to avoid using enum in Content Model because
- Normal enum will cause extra bundle increase
- Const enum has some problem when used by isolated typescript project

So string literal type is the best here.

Reply:
That is good to know!!